airflow/utils/log/file_task_handler.py

```diff
@@ -116,7 +116,7 @@ def _read(self, ti, try_number, metadata=None):
 
         if os.path.exists(location):
             try:
-                with open(location) as file:
+                with open(location, encoding="utf-8") as file:
```

Comment:
Maybe we should do `error="replace"` or something so a log written in another encoding can still be partially read. I’m suspecting this input-output encoding mismatch only happens on Python 3.7+, when Python starts automatically coerce the default legacy C to UTC-8 ([PEP 538](https://www.python.org/dev/peps/pep-0538/)). If this is the cause, setting `utf-8` here is a correct move, but we should still provide a fallback for weird platforms that need/want to use a platform-specific encoding. Perhaps a new config to explicitly set what encoding to use here?

Reply:
Makes sense. I feel, we should use 'surrogateescape' as replacement as it will provide the exact bytes that caused the issue, making it more specific for user if they desire to do so.
And yes, it would definetly be better to have a config to set the encoding.
I have updated the PR with replacement as surrogate character as of now and created a feature request #17993 and would like to work on this requirement.